src/pb_stub.cc

```diff
@@ -837,6 +857,135 @@ Stub::GetOrCreateInstance()
   return Stub::stub_instance_;
 }
 
+void
+Stub::LaunchLogRequestThread()
+{
+  log_thread_ = true;
+  log_monitor_ = std::thread(&Stub::ServiceLogRequests, this);
+}
+
+void
+Stub::TerminateLogRequestThread()
+{
+  log_thread_ = false;
+  {
+    bi::scoped_lock<bi::interprocess_mutex> guard{log_message_mutex_};
+    log_request_buffer.push(DUMMY_MESSAGE);
+  }
+  log_message_cv_.notify_all();
+  log_monitor_.join();
+}
+void
+Stub::EnqueueLogRequest(std::unique_ptr<PbLog>& log_ptr)
+{
+  {
+    bi::scoped_lock<bi::interprocess_mutex> guard{log_message_mutex_};
+    log_request_buffer.push(std::move(log_ptr));
+  }
+  log_message_cv_.notify_one();
+}
+
+void
+Stub::ServiceLogRequests()
+{
+  while (log_thread_) {
+    bi::scoped_lock<bi::interprocess_mutex> guard{log_message_mutex_};
```

Comment:
Since this mutex is not interprocess, you should use the std library mutex and condition variable. https://en.cppreference.com/w/cpp/thread/mutex